core/extraction/towns_fund_round_three.py

```diff
@@ -956,9 +957,10 @@ def extract_footfall_outcomes(df_input: pd.DataFrame, project_lookup: dict, prog
     if invalid_projects := relevant_projects - set(project_lookup.keys()):
         raise ValidationError(
             [
-                vf.InvalidOutcomeProjectFailure(invalid_project=project, section="Footfall Indicator", row_indexes=None)
-                # TODO: add index for footfalls
-                for project in invalid_projects
+                vf.InvalidOutcomeProjectFailure(
+                    invalid_project=row["Relevant Project(s)"], section="Footfall Indicator", row_indexes=[idx + 7]
+                )  # extra +5 for Outcome -> Project row
+                for idx, row in footfall_df.loc[footfall_df["Relevant Project(s)"].isin(invalid_projects)].iterrows()
```

Comment:
unclear as to why it says `+5` in the comment but `+7` in the code?

Reply:
Because the +5 represents "Footfall Indicator" and "Relevant Project(s)" being 5 rows apart, and the +2 is just generic for the spreadsheet. I should have made that clearer, will amend.

Reply:
Can we make it even more explicit by doing `idx + 2 + 5` and reference why we do the `+2` here as well?

Could you also just add a general comment about why we're checking this and raising a Failure so it's super clear? It deviates from our usual validation so worth explaining in a comment I think otherwise it could be hard for someone without the context to know what's going on here.

Reply:
Think it's still missing a general comment about raising these Validation Failures

Reply:
specifically, why we raise one here instead of during validation